Bottom: 808ac23afd96a09fcda40f36c9c369b7cab72b1f
Top:    50de165746c0f1b7b4cea9e6514cc651306b9cbc
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-01-07 23:06:49 +0000

Fix layer on/off rendering

I'm not certain this is correct at all. It works around failure
to switch layer visibility on / off after

commit 9768e060fad7bc3dfc366da76ea1db8154005018
"Fix layer visibility logic for some boards."

After which, copper layers on the component / solder group
would still be drawn unless the silk layer was also turned off.

This appears to be due to the GL HID's custom drawing routines
not checking the return of the (HID)->set_layer function in certain
instances.

This fix is almost certainly WRONG.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index bb4515b..40c50b7 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1419,7 +1419,7 @@ DrawLayerGroup (int group, const BoxType * screen)
         strcmp (Layer->Name, "route") == 0)
       rv = 0;
 
-    if (layernum < max_copper_layer /*&& Layer->On*/) {
+    if (layernum < max_copper_layer && Layer->On) {
 
       if (!first_run)
         gui->set_layer (0, group, 0);
